File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int neg = 0;
        
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[i].length; j++){
                if(grid[i][j] < 0) neg++;
            }
        }
        return neg;
    }
}

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.swift
class Solution {
    func countNegatives(_ grid: [[Int]]) -> Int {
        var res = 0
        var len = grid.count
        var i = 0
        var j = grid[0].count - 1
        
        while i < len && j >= 0 {
            if grid[i][j] < 0 {
                res += len - i
                j -= 1
            } else {
                i += 1
            }
        }
        
        return res
    }
}
